### DISCUSSION TITLE
Report number of files scanned in SARIF reports

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [detekt/detekt#7669] [Ideas]
I use both CodeQL and Detekt for code scanning in my GitHub repo. CodeQL is able to report how many files it scanned, like so:

![Screen Shot 2024-09-24 at 11 09 57 AM](https://github.com/user-attachments/assets/a4956fdb-a8be-4d5c-a967-a41f753bf04c)

I don't think Detekt does this at present. I think it should include the file count in generated reports.

**Comment by BraisGabin** [comment]
Didn't we report the number of files? I think that in the html we do. So if you want it in other format you can create your own report, detekt's api will give you that value.

**Comment by JordanLongstaff** [comment]
I use CodeQL's [upload-sarif action](https://github.com/github/codeql-action) to report the issues detected in GitHub. CodeQL doesn't have an uploader for HTML reports that I can find.

I see that the HTML reports do include the metrics of scanned files, classes etc. but for compatibility with GitHub's code scanning tools, it'll need to be included in SARIF reports too.

Unless you have a better solution. [Here](https://github.com/JordanLongstaff/ArtemisAgent/blob/main/.github/workflows/detekt.yml) is what my Detekt GitHub Actions workflow looks like.

**Reply by BraisGabin** [reply]
Oh! I see. That's another topic. We could add the number of files on the sarif report. Do you know where in the sarif spec is that defined? Or where does codeQL store it?

If you know that information you can open a new issue so we can add that. Or, if you have the time, you can create a PR adding it directly to our sarif report implementation.

**Reply by JordanLongstaff** [reply]
I retrieved a [CodeQL SARIF result file](https://github.com/user-attachments/files/18613556/java.sarif.txt) from my repo's workflow. The files that are expected to be extracted are marked under the `cli/expected-extracted-files/kotlin` notification, and the files that are actually extracted are marked under `java/diagnostics/successfully-extracted-files`.

**Reply by BraisGabin** [reply]
I don't get what `cli/expected-extracted-files/kotlin` and `java/diagnostics/successfully-extracted-files` provides here. I can't find a count of files related with those strings.

But what we could inform is `runs.artifacts`. And list there all the files that were reviewed. Would that help with your use case?

**Reply by BraisGabin** [reply]
Also, detekt can only say how many files it reviewed. Detekt can't say how many files exist. Detekt receives a list of files to review, it doesn't check anything else.
